config/environments/production.rb

```diff
@@ -28,7 +28,8 @@
   # config.assets.css_compressor = :sass
 
   # Do not fallback to assets pipeline if a precompiled asset is missed.
-  config.assets.compile = false
+  config.assets.compile = true
+  config.serve_static_files = true
```

Comment:
This reminds me of Heroku's 12-factor apps. If you're interested, you could look at the factors and potentially implement them here (there's a gem that does a lot of the work).

https://12factor.net/